src/api/store/vendor.py

```diff
@@ -6,9 +6,24 @@
 from django.db.models import Q
 from .utils import get_community_or_die, get_admin_communities
 from _main_.utils.context import Context
+from database.utils.constants import SHORT_STR_LEN
 from sentry_sdk import capture_message
 from typing import Tuple
-import random
+from .utils import get_new_title
+
+def _copy_title(old_title):
+  # return first unique title
+  # Community will be None at first
+  new_title = get_new_title(None, old_title)
+  version = 0
+  while version<1000:
+    suffix = "-Copy%d" % version
+    newlen = min(len(new_title), SHORT_STR_LEN - len(suffix))
+    title = new_title[0:newlen] + suffix
+    if not Vendor.objects.filter(name=title).first():
+      return title
+    version += 1  
```

Comment:
Could we explore other options here?  
In the worst case this function seems like it will query the DB 1000 times just to create a title.

Also i am not sure of the goal exactly

Reply:
The goal is to pick the first available title that doesn't duplicate one already used.  I don't like the random number solution, because from the community admin perspective it looks like a hack.  The usual case is that a community admin copies an action/event/vendor and the first try ("-Copy0") works, then they edit the action/event/vendor to be for their community and change the name to what they want.  In a rare case, someone might Copy something several times without changing the name, and we'd have a few copies.  Not 1000, and we can have the chron job go through monthly and delete extraneous copies of actions/events/vendors that aren't connected with any communities or published.
Certainly could be better solutions, but we don't want to go back to the random numbers.

Reply:
Hmmm i don't think we want this either.
First the query itself is filtering by a column that is not indexed. 
Second, we will do this in a loop that could possibly go a 1000 times.
Considering these are all network calls and will incur overheads, i don't think we should go with this approach either.  I'd say until we have a better solution that random numbers, let's not do anything to compromise on performance.